scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+      }
+    }
+  }
+  return licenseSet
+}
+
+private fun updateMavenDependenciesList(
+  latestDependenciesList: List<MavenDependency>,
+  finalLicensesSet: Set<License>
+): MutableList<MavenDependency> {
+  val finalUpdatedList = mutableListOf<MavenDependency>()
+
+  latestDependenciesList.forEach { mavenDependency ->
+    val updateLicenseList = mutableListOf<License>()
+    var numberOfLicensesToBeProvidedManually = 0
+    mavenDependency.licenseList.forEach { license ->
+      val updatedLicense = finalLicensesSet.find { it.primaryLink == license.primaryLink }
+      if (updatedLicense != null) {
+        updateLicenseList.add(updatedLicense)
+        if (updatedLicense.primaryLinkType == PrimaryLinkType.NEEDS_INTERVENTION ||
+          updatedLicense.primaryLinkType == PrimaryLinkType.INVALID_LINK
+        ) {
+          numberOfLicensesToBeProvidedManually++
+        }
+      } else {
+        if (license.primaryLinkType == PrimaryLinkType.NEEDS_INTERVENTION ||
+          license.primaryLinkType == PrimaryLinkType.INVALID_LINK
+        ) {
+          numberOfLicensesToBeProvidedManually++
+        }
+        updateLicenseList.add(license)
+      }
+    }
+    val dependency = MavenDependency.newBuilder()
+      .setArtifactName(mavenDependency.artifactName)
+      .setArtifactVersion(mavenDependency.artifactVersion)
+      .addAllLicense(updateLicenseList)
+      .setOriginOfLicense(OriginOfLicenses.UNKNOWN)
+
+    if (updateLicenseList.isNotEmpty()) {
+      if (numberOfLicensesToBeProvidedManually == updateLicenseList.size) {
+        dependency.originOfLicense = OriginOfLicenses.MANUAL
+      } else if (numberOfLicensesToBeProvidedManually == 0) {
+        dependency.originOfLicense = OriginOfLicenses.ENTIRELY_FROM_POM
+      } else if (numberOfLicensesToBeProvidedManually != updateLicenseList.size) {
+        dependency.originOfLicense = OriginOfLicenses.PARTIALLY_FROM_POM
+      }
+    }
+    finalUpdatedList.add(dependency.build())
+  }
+  return finalUpdatedList
+}
+
+/** Retrieves the list of [MavenDependency] from maven_dependencies.textproto. */
+private fun retrieveMavenDependencyList(): List<MavenDependency> {
+  return getProto(
+    "maven_dependencies.pb",
+    MavenDependencyList.getDefaultInstance()
+  ).mavenDependencyList.toList()
+}
+
+/**
+ * Helper function to parse the textproto file to a proto class.
+ *
+ * @param textProtoFileName name of the textproto file to be parsed
+ * @param proto instance of the proto class
+ * @return proto class from the parsed textproto file
+ */
+private fun getProto(
+  textProtoFileName: String,
+  proto: MavenDependencyList
+): MavenDependencyList {
+  val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+  val builder = proto.newBuilderForType()
+  val protoObject = FileInputStream(protoBinaryFile).use {
+    builder.mergeFrom(it)
+  }.build() as MavenDependencyList
+  return protoObject
+}
+
+fun parseArtifactName(artifactName: String): String {
```

Comment:
Here & below: these seem to be helpers and thus should be private (generally I think everything except main() should probably be private in this file).

Reply:
Done.

Reply:
That's actually not what we want to do. When testing a component, you should always test it through its public API. Exposing implementation pieces to tests is fragile: if the implementation changes, the tests may fail. Tests should only fail if the contract they're verifying no longer matches the contract by the implementation. In many cases we change implementations without changing the contract (the most common scenarios being cases when we refactor the code). In these situations, the tests ought to never even be updated or only updated superficially (such as if the APIs themselves are slightly refactored without changing meaning).

Reply:
Acknowledged. Made all the functions private.

Reply:
Verfied.